src/inputs/internal/ListBox.tsx

```diff
@@ -24,17 +24,24 @@ export function ListBox<O, V extends Key>(props: ListBoxProps<O, V>) {
     state,
     compact,
     listBoxRef,
-    // Choosing `273` as a defined max-height. `42px` is the min-height of each option, so this allows
-    // 6.5 options in view at a time (doing `.5` so the user can easily tell if there are more).
-    maxListHeight = 273,
     selectedOptions,
     getOptionLabel,
     getOptionValue,
     contrast = false,
+    positionProps,
     ...otherProps
   } = props;
   const { listBoxProps } = useListBox({ disallowEmptySelection: true, ...otherProps }, state, listBoxRef);
 
+  const positionMaxHeight = positionProps.style?.maxHeight;
+  // The maxListHeight will be based on the value defined by the positionProps returned from `useOverlayPosition` (which will always be a defined as a `number` based on React-Aria's `calculatePosition`).
+  // If `maxHeight` is set use that, otherwise use `273` as a default (`42px` is the min-height of each option, so this allows
+  // 6.5 options in view at a time (doing `.5` so the user can easily tell if there are more).
+  const maxListHeight = positionMaxHeight && typeof positionMaxHeight === "number" ? positionMaxHeight : 273;
+  // We define some spacing between the ListBox and the trigger element, and depending on where the list box renders (above or below) we need to adjust the spacing.
+  // We can determine if the position was flipped based on what style is defined, `top` (for positioned below the trigger), and `bottom` (for above the trigger).
+  // The above assumption regarding `top` and `bottom` is true as long as we use `bottom` as our `OverlayPosition.placement` (set in SelectFieldBase).
```

Comment:
So the `placement: bottom` down below doesn't actually mean "always bottom"? I.e. it sounds like you're still detecting that sometimes it could be above.

Reply:
Right, we also set `shouldFlip: true`. I'll add that into the comment.